src/reformed.js

```diff
@@ -45,11 +46,26 @@ const makeWrapper = (middleware) => (WrappedComponent) => {
       }
     }
 
-    bindInput = (name) => {
+    bindInput = (name, opts) => {
+      const cfg = assign({
+        updateOn: 'onChange',
+        debounce: 0
+      }, opts)
+
+      const { updateOn } = cfg
+      const asyncHandler = (fn) => (e) => {
+        e.persist()
```

Comment:
Maybe I have to pull this to try it,  but shouldn't this be more like:

```js
const debouncedUpdater = debounce(...)
const asyncUpdater = (e) => {
  e.persist()
  debouncedUpdater(e)
}
```

Maybe your implementation works and my code comprehension is lacking, but it would seem like you'd need to _immediately_ persist the event, and then that way the debounced event handler 
(called x ms later) can receive the persisted event.

I _think_ yours works just as well, but if I'm understanding it correctly the `e.persist()` is unnecessary due to how the debouncing and event handler is working. Your handler only gets called once debounce allows it to be, but at that point the event you receive will be immediately passed to the original handler without any further delay.

I think either implementation is likely fine (and yours probably better; mine would persist a whole bunch of ignored events) but I'd have to run the code before I could comment further.